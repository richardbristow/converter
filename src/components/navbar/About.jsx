import React from 'react';

const About = () => (
  <div>
    <h1>About</h1>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit.
       Delectus deleniti nam beatae sint voluptas ea blanditiis,
       sequi itaque animi vero nisi aperiam necessitatibus
       mollitia est voluptates nemo sit at temporibus?
    </p>
    <h2>Credits</h2>
    <ul>
      <li>react</li>
      <li>math.js</li>
      <li>fixe</li>
    </ul>
  </div>
);


export default About;
