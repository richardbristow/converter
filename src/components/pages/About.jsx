import React from 'react';
import StyledNavPage from '../shared/StyledNavPage';

const About = () => (
  <StyledNavPage>
    <h2>About</h2>
    <p>
      Converter is a webapp to convert between various units of measurement.
    </p>
    <p>
      It was built mainly as a way to learn various web technologies, one of
      them being css-grid, therefore it most likely will not render in IE
      correctly.
    </p>
    <p>
      If you are interested in seeing the code it&apos;s available on&nbsp;
      <a href="https://github.com/richardbristow/converter" title="Github">
        Github
      </a>
      .
    </p>
    <br />
    <h3>Libraries / API&apos;s used</h3>
    <ul>
      <li>
        <a href="https://reactjs.org" title="React">
          React
        </a>
      </li>
      <li>
        <a href="http://mathjs.org/" title="MathJs">
          Math.js
        </a>
      </li>
      <li>
        <a href="https://fixer.io/" title="FixerIO">
          Fixer.io
        </a>
      </li>
    </ul>
    <br />
    <h3>Icons</h3>
    <div>
      <span>
        Flag icons made by&nbsp;
        <a href="https://www.flaticon.com/authors/twitter" title="Twitter">
          Twitter
        </a>
        &nbsp;from&nbsp;
        <a href="https://www.flaticon.com" title="Flaticon">
          Flaticon.com
        </a>
      </span>
    </div>
    <div>
      <span>
        Bitcoin icon made by&nbsp;
        <a
          href="https://www.flaticon.com/authors/pixel-buddha"
          title="Pixel Buddha"
        >
          Pixel Buddha
        </a>
        &nbsp;from&nbsp;
        <a href="https://www.flaticon.com/" title="Flaticon">
          Flaticon.com
        </a>
      </span>
    </div>
    <div>
      <span>
        Mass icon made by&nbsp;
        <a href="http://www.freepik.com" title="Freepik">
          Freepik
        </a>
        &nbsp;from&nbsp;
        <a href="https://www.flaticon.com/" title="Flaticon">
          Flaticon.com
        </a>
      </span>
    </div>
    <div>
      <span>
        Temperature icon made by&nbsp;
        <a
          href="https://www.flaticon.com/authors/hirschwolf"
          title="hirschwolf"
        >
          hirschwolf
        </a>
        &nbsp;from&nbsp;
        <a href="https://www.flaticon.com/" title="Flaticon">
          Flaticon.com
        </a>
      </span>
    </div>
    <div>
      <span>
        All other icons from&nbsp;
        <a href="https://iconmonstr.com/about/#creator">Iconmonstr</a>
      </span>
    </div>
  </StyledNavPage>
);

export default About;
