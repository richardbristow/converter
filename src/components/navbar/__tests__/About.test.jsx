import React from 'react';
import { shallow } from 'enzyme';
import About from '../About';

describe('About', () => {
  it('should render the About page', () => {
    const wrapper = shallow(<About />);
    expect(wrapper).toMatchSnapshot();
  });
});
