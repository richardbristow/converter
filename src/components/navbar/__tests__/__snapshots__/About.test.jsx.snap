// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should render the About page 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <About />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          About
</h1>,
        <p>
          Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?
</p>,
        <h2>
          Credits
</h2>,
        <ul>
          <li>
                    react
          </li>
          <li>
                    math.js
          </li>
          <li>
                    fixe
          </li>
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "About",
        },
        "ref": null,
        "rendered": "About",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?",
        },
        "ref": null,
        "rendered": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Credits",
        },
        "ref": null,
        "rendered": "Credits",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              react
</li>,
            <li>
              math.js
</li>,
            <li>
              fixe
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "react",
            },
            "ref": null,
            "rendered": "react",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "math.js",
            },
            "ref": null,
            "rendered": "math.js",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "fixe",
            },
            "ref": null,
            "rendered": "fixe",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            About
</h1>,
          <p>
            Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?
</p>,
          <h2>
            Credits
</h2>,
          <ul>
            <li>
                        react
            </li>
            <li>
                        math.js
            </li>
            <li>
                        fixe
            </li>
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "About",
          },
          "ref": null,
          "rendered": "About",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?",
          },
          "ref": null,
          "rendered": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Delectus deleniti nam beatae sint voluptas ea blanditiis, sequi itaque animi vero nisi aperiam necessitatibus mollitia est voluptates nemo sit at temporibus?",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Credits",
          },
          "ref": null,
          "rendered": "Credits",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                react
</li>,
              <li>
                math.js
</li>,
              <li>
                fixe
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "react",
              },
              "ref": null,
              "rendered": "react",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "math.js",
              },
              "ref": null,
              "rendered": "math.js",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "fixe",
              },
              "ref": null,
              "rendered": "fixe",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
